src/utils/cancelRequests.ts

```diff
@@ -14,13 +14,22 @@ export type RequestConfiguration = {
 export class CancelToken {
   protected token: CancelTokenAxios | null = null;
   protected source: CancelTokenSource | null = null;
+  protected cacheKey: string | null = null;
 
   public constructor() {
     this.source = axios.CancelToken.source();
     this.token = this.source.token;
   }
 
+  public setCancelTokenCacheKey(cacheKey: string | null): void {
+    this.cacheKey = cacheKey;
```

Comment:
A single `CancelToken` can be used for multiple requests. So I think this should be an array with all the `cacheKeys` inside, and then all would be removed on cancelling.

I believe this fails:

```
it('handles multiple requests with the same cancel token', async () => {
    let requestError = null;
    const { requestPromise, thenFn, catchFn, cancelToken, mockedResponse } = createRequestPromise(
      true,
      err => (requestError = err),
    );

    const {layer, getMapParams, mockedResponse: mockedResponse2} = constructFixtureGetMap()

    mockNetwork.onPost().replyOnce(200, mockedResponse);
    mockNetwork.onPost().replyOnce(200, mockedResponse2);

    setAuthToken("EXAMPLE_TOKEN");

    await Promise.all([requestPromise, layer.getMap(getMapParams, ApiType.PROCESSING, {cancelToken: cancelToken, cache: {
      expiresIn: 60,
      targets: [CacheTarget.MEMORY],
    }}).catch((err:any) => console.log(err)), setTimeout(() => cancelToken.cancel(), 10)]);

    expect(cacheableRequestsInProgress.size).toBe(0);
  });
  ```

Reply:
Thanks for spotting this, I didn't think about that. I added test and changed to support multiple cacheKeys.